axelor-account/src/main/java/com/axelor/apps/account/service/BudgetService.java

```diff
@@ -152,4 +187,30 @@ public void checkSharedDates(Budget budget) throws AxelorException {
       }
     }
   }
+
+  @Transactional
+  public void validate(Budget budget) {
+    budget.setStatusSelect(BudgetRepository.STATUS_VALIDATED);
+  }
+
+  @Transactional
+  public void draft(Budget budget) {
+    budget.setStatusSelect(BudgetRepository.STATUS_DRAFT);
+  }
+
+  public void updateBudgetLinesFromInvoice(Invoice invoice) {
+    List<InvoiceLine> invoiceLineList = invoice.getInvoiceLineList();
+
+    if(invoiceLineList == null){
+      return;
+    }
+
+    invoiceLineList.forEach(invoiceLine ->
```

Comment:
Maybe it's clearer to use flatMap and map here, and add null check at least for the list. I think budget is not required in db, so you must check if the budget is null too (filter(Objects::nonNull))